src/toil/job.py

```diff
@@ -1282,7 +1331,7 @@ def __init__(self, job):
         :param toil.job.Job job: the job to encapsulate.
         """
         # Giving the root of the subgraph the same resources as the first job in the subgraph.
-        Job.__init__(self, disk=job.disk, memory=job.memory, cores=job.cores)
+        Job.__init__(self, **job._resources)
```

Comment:
Sounds like we need `_resources` and `_effectiveResources`.

BTW, can we call those properties `_requirements` and `_effectiveRequirements` instead? That would avoid the conflict with the use of the term `resource` for hot deployment. It would also be consistent with PromisedRequirement etc. …